python/action-server/state_machine.py

```diff
@@ -0,0 +1,78 @@
+        self.session.put("/state", self.state)
+        
+
+    def trigger(self, trigger_method):
+        try:
+            self.session.put("/trigger/"+trigger_method, {'timestamp':time.time(), 'triggered':trigger_method})
+        except Exception as e:
+            self.session.put("/error/"+trigger_method, {'timestamp':time.time(), 'Error': e})
+
+
+class Unhealthy(HierarchicalMachine, Session_state):
+    def __init__(self):
+        states = [{"name":'aborted', "on_enter":["aborted"]},
+                    {"name":"awaitingclearanceerr", 'on_enter':['cleared']},
+                    {"name":"cleared", 'on_enter':'brokenwithoutholdings'},
+                    {"name":"brokenwithholdings", 'on_enter':'dead'},
+                    {"name":"brokenwithoutholdings", 'on_enter':'dead'},
+                    {"name":"dead"}
+                    ]
+        transitions = [{"trigger":"aborted", "source":"aborted", "dest":"awaitingclearanceerr"},
+                        {"trigger":"awaitingclearanceerr", "source":"clearancetimeout", "dest":"awaitingclearanceerr"},
+                        {"trigger":"cleared", "source":"awaitingclearanceerr", "dest":"cleared"},
+                        {"trigger":"brokenwithoutholdings", "source":"cleared", "dest":"brokenwithoutholdings"},
+                        {"trigger":"brokenwithholdings", "source":"awaitingclearanceerr", "dest":"brokenwithholdings"},
+                        {"trigger":"dead", "source":"brokenwithholdings", "dest":"dead"},
+                        {"trigger":"dead", "source":"brokenwithoutholdings", "dest":"dead"}]
+        super().__init__(states=states, transitions=transitions, initial="awaitingclearanceerr", queued=QUEUED)
+
+class Healthy(HierarchicalMachine, Session_state):
+    def __init__(self):
+        aborted = Unhealthy()
+        states = [{"name":'idle', 'on_enter':[self.state_(), 'start']},
+                    {"name":"busy"},
+                    {"name":"aborted", "children": aborted},
+                    {"name":"done"},
+                    {"name":"clearancetimeout", "children": aborted},
+                    {"name":"awaitingclearance"}]
+
+        transitions = [{'trigger':'start', 'source':'idle', 'dest':'busy'},
+                        {"trigger":"abort", "source":"busy", "dest":"aborted"},
+                        {"trigger":"done", "source":"busy", "dest":"done"},
+                        {"trigger":"awaitingclearance", "source":"done", "dest":"awaitingclearance"},
+                        {"trigger":"clearancetimeout", "source":"awaitingclearance", "dest":"clearancetimeout"},
+                        {"trigger":"idle", "source":"awaiting_clearance", "dest":"idle"}]
+        super().__init__(states=states, transitions=transitions, initial="idle", queued=QUEUED)
+
+
+class StateMachine(HierarchicalMachine, MarkupMachine, Session_state):
+    def __init__(self):
+        healthy = Healthy()
+        states = ["idle", {"name":'healthy', 'children':healthy}]
```

Comment:
there are only 2 top level states:
```
initial to healthy
healthy to unhealthy
```

thats it

Reply:
It is doing same but unhealthy is not needed in the top level. At top level, it will go from `idle to healthy` and once it is in healthy state then it can go to unhealthy if needed or can came back to idle.